src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -42,144 +34,129 @@ enum Spells
-Position const ADD_POS_3 = {4017.738037f, -3363.478016f, 115.057274f, 0.723313f};
-Position const ADD_POS_4 = {4048.877197f, -3363.223633f, 115.054253f, 3.627735f};
-Position const ADD_POS_5 = {4051.777588f, -3350.893311f, 115.055351f, 3.066176f};
-Position const ADD_POS_6 = {4048.375977f, -3339.966309f, 115.055222f, 2.457497f};
+enum BaronRivendareEvents
+{
+    EVENT_SPELL_SHADOWBOLT      = 1,
+    EVENT_SPELL_CLEAVE          = 2,
+    EVENT_SPELL_MORTALSTRIKE    = 3,
+    EVENT_SUMMON_SKELETON       = 4
+};
 
-class boss_baron_rivendare : public CreatureScript
+Position const PosSkeleton[6] =
 {
-public:
-    boss_baron_rivendare() : CreatureScript("boss_baron_rivendare") { }
+    { 4017.403809f, -3339.703369f, 115.057655f, 5.487860f },
+    { 4013.189209f, -3351.808350f, 115.052254f, 0.134280f },
+    { 4017.738037f, -3363.478016f, 115.057274f, 0.723313f },
+    { 4048.877197f, -3363.223633f, 115.054253f, 3.627735f },
+    { 4051.777588f, -3350.893311f, 115.055351f, 3.066176f },
+    { 4048.375977f, -3339.966309f, 115.055222f, 2.457497f }
+};
 
-    CreatureAI* GetAI(Creature* creature) const override
+struct boss_baron_rivendare : public BossAI
+{
+    boss_baron_rivendare(Creature* creature) : BossAI(creature, TYPE_BARON), _raiseDead(false) { }
+
+    void Reset() override
     {
-        return GetStratholmeAI<boss_baron_rivendareAI>(creature);
+        if (instance->GetData(TYPE_RAMSTEIN) == DONE)
+            instance->SetData(TYPE_BARON, NOT_STARTED);
+
+        _raiseDead = false;
+        events.Reset();
     }
 
-    struct boss_baron_rivendareAI : public ScriptedAI
+    // can't use entercombat(), boss' dmg aura sets near
+    // players in combat, before entering the room's door
+    void AttackStart(Unit* /*who*/) override
     {
-        boss_baron_rivendareAI(Creature* creature) : ScriptedAI(creature)
-        {
-            Initialize();
-            instance = me->GetInstanceScript();
-        }
-
-        void Initialize()
-        {
-            ShadowBolt_Timer = 5000;
-            Cleave_Timer = 8000;
-            MortalStrike_Timer = 12000;
-            //        RaiseDead_Timer = 30000;
-            SummonSkeletons_Timer = 34000;
-        }
+        if (instance->GetData(TYPE_BARON) == NOT_STARTED)
+            instance->SetData(TYPE_BARON, IN_PROGRESS);
 
-        InstanceScript* instance;
+        events.ScheduleEvent(EVENT_SPELL_SHADOWBOLT, 5s);
+        events.ScheduleEvent(EVENT_SPELL_CLEAVE, 8s);
+        events.ScheduleEvent(EVENT_SPELL_MORTALSTRIKE, 12s);
+        events.ScheduleEvent(EVENT_SUMMON_SKELETON, 15s);
+    }
 
-        uint32 ShadowBolt_Timer;
-        uint32 Cleave_Timer;
-        uint32 MortalStrike_Timer;
-        //    uint32 RaiseDead_Timer;
-        uint32 SummonSkeletons_Timer;
+    void JustSummoned(Creature* summoned) override
+    {
+        if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
+            summoned->AI()->AttackStart(target);
```

Comment:
summons.Summon(summoned); and despawn them on reset

Reply:
@Riztazz are you sure? I mean it was a long time ago but I remember that they did not despawn instantly after a wipe... they ran back to their initial positions and just despawn after some time

Reply:
Boss adds are usually despawned when boss itself evades, pretty much the case everywhere, but maybe its different in here :P